comp/core/secrets/secretsimpl/fetch_secret.go

```diff
@@ -41,30 +37,42 @@ func (b *limitBuffer) Write(p []byte) (n int, err error) {
 	return b.buf.Write(p)
 }
 
-func execCommand(inputPayload string) ([]byte, error) {
+func (r *secretResolver) execCommand(inputPayload string) ([]byte, error) {
+	if r.commandHookFunc != nil {
+		return r.commandHookFunc(inputPayload)
+	}
+	commandTimeout := r.backendTimeout
+	if commandTimeout == 0 {
+		commandTimeout = SecretBackendTimeoutDefault
+	}
+	responseMaxSize := r.responseMaxSize
+	if responseMaxSize == 0 {
+		responseMaxSize = SecretBackendOutputMaxSizeDefault
+	}
```

Comment:
I think this should be done in `Configure` so we don't need to re-check for default at every exec.

Reply:
I had thought it was better to make Configure optional, based upon how tests were using secrets, but I just fixed the tests instead.